[Shanice Ng] iP

shaniceng opened the pull request:
## DukePro
> “Your mind is for having ideas, not holding them.” – David Allen ([source](https://dansilvestre.com/productivity-quotes/)) 

DukePro frees your mind of having to remember things you need to do. It's,
- text-based
- easy to learn
- ~~FAST~~ *SUPER* FAST to use

All you need to do is,
* download it from [here](https://nus-cs2103-ay2223s1.github.io/website/schedule/week4/project.html).
* double-click it.
* add your tasks.
* let it manage your tasks for you 😉

And it is **FREE**!

Features:
- [x] Managing tasks
- [ ] Managing deadlines (coming soon)
- [ ] Reminders (coming soon)

If you Java programmer, you can use it to practice Java too. Here's the `main` method:
```
public class Main {
    public static void main(String[] args) {
        Application.launch(MainApp.class, args);
    }
}
```

Commit:
Add Gradle support

shaniceng committed:
level 1 complete

shaniceng committed:
Revert "level 1 complete"

This reverts commit 3ff0af149b1698f2e3665f76d8ed12d6a42feb4d.

shaniceng committed:
level 1 complete

shaniceng committed:
level 2 complete

shaniceng committed:
level 3 complete

shaniceng committed:
level 4 complete

shaniceng committed:
edit level 4

shaniceng committed:
level 5 complete

shaniceng committed:
level 6 complete

shaniceng committed:
Update DukeException and include functions to handle each input keyword

shaniceng committed:
Add Level-7

shaniceng committed:
Merge branch 'branch-Level-7'

shaniceng committed:
Add Level-8

shaniceng committed:
Merge branch 'branch-Level-8'

shaniceng committed:
Add A-MoreOOP

shaniceng committed:
Add A-Packages

shaniceng committed:
Merge branch 'add-gradle-support' of https://github.com/shaniceng/ip

shaniceng committed:
Add A-Gradle

shaniceng committed:
Add 3 test methods for A-JUnit

shaniceng committed:
Add A-Jar

shaniceng committed:
Revert "Add A-Jar"

This reverts commit 03fd7b0d3794f72d6b51b53d18876ff0861f62aa.

shaniceng committed:
Add JavaDoc comments to the code

shaniceng committed:
Tweak the code to comply with a coding standard

shaniceng committed:
Add Level-9

shaniceng committed:
Merge branch 'branch-A-JavaDoc'

shaniceng committed:
Merge branch 'branch-A-CodingStandard'

# Conflicts:
#	src/main/java/Duke/Ui.java

shaniceng committed:
Merge branch 'branch-Level-9'

# Conflicts:
#	src/main/java/Duke/Parser.java
#	src/main/java/Duke/TaskList.java

shaniceng committed:
Edit Date Time format to include time and edit naming convention

Donovan9617 commented on src/test/java/Tasks/DeadlineTest.java:
```diff
@@ -0,0 +1,17 @@
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+
+public class DeadlineTest {
```
Perhaps you forgot to add java docs for test class and method here?

shaniceng replied:
yes, I did not add java docs for all public classes and methods. Thanks for highlighting :)

Donovan9617 commented on src/test/java/Tasks/EventTest.java:
```diff
@@ -0,0 +1,17 @@
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+
+public class EventTest {
```
Perhaps you forgot to add java docs for test class and method here too?

Donovan9617 commented on src/test/java/Tasks/TodoTest.java:
```diff
@@ -0,0 +1,13 @@
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+
+public class TodoTest {
```
Perhaps you forgot to add java docs for test class and method here too?

Donovan9617 commented on src/main/java/Duke/Ui.java:
```diff
@@ -0,0 +1,106 @@
+     * @throws DukeException which shows an error if there is no more lines
+     *         left in the user input
+     */
+    public String readCommand() throws DukeException {
+        if (scanner.hasNextLine())
+            return scanner.nextLine();
+        else {
+            throw new DukeException(Constants.NO_MORE_LINES);
+        }
+    }
```
This works yay, but can also consider try-catch cos is normally used for error handling

shaniceng replied:
okay! Thanks for the suggestion :)

Donovan9617 commented on src/main/java/Command/TodoCommand.java:
```diff
@@ -0,0 +1,34 @@
+import Duke.Ui;
+import Tasks.Todo;
+
+public class TodoCommand extends Command{
```
oops forgot gap between Command and { haha

shaniceng replied:
oops sorry, I missed out on that. Thanks for highlighting it :)

Donovan9617 commented on src/main/java/Duke/Duke.java:
```diff
@@ -0,0 +1,53 @@
+     *
+     * @param filePath which contains the relative path of the saved task list
+     */
+    public Duke(String filePath) {
+        ui = new Ui();
+        storage = new Storage(filePath);
+        tasks = new TaskList(storage.readFile());
+    }
```
hmm try-catch may be better here cos must handle case where filePath may not exist

shaniceng replied:
Oh okay! Thanks for the suggestion :)

Donovan9617 commented on src/main/java/Duke/Parser.java:
```diff
@@ -0,0 +1,98 @@
+    public static Command parse(String command) throws DukeException {
+        String[] input = command.split(" ", 2);
+        DateTimeFormatter formatter = DateTimeFormatter.ofPattern("dd-MMM-yyyy HH:mm");
+        switch (input[0]) {
```
May be nice to give input[0] a variable name too

shaniceng replied:
Okay! Thanks for highlighting :)

Donovan9617 reviewed (commented):
Very nice! Good job for being organised and neat with the code LGTM :)

shaniceng committed:
Add GUI
Add more test methods for A-Junit
Add more javaDoc comments for all public classes and methods
Edit README.md to include pictures taken from Freepik and rawpixel.com

shaniceng committed:
Revert "Add GUI"

This reverts commit b2468cd9146a09cf469659c936e858078ca6de37.

shaniceng committed:
Revert "Add GUI"

This reverts commit b2468cd9146a09cf469659c936e858078ca6de37.

shaniceng committed:
Add GUI
Add more test methods for A-Junit
Add more javaDoc comments for all public classes and methods
Edit README.md to include pictures taken from Freepik and rawpixel.com

shaniceng committed:
Merge branch 'branch-Level-10'

# Conflicts:
#	README.md
#	data/tasks.txt
#	src/main/java/Command/DeadlineCommand.java
#	src/main/java/Command/ExitCommand.java
#	src/main/java/Command/FindCommand.java
#	src/main/java/Command/TodoCommand.java
#	src/main/java/Duke/Ui.java

chngchngchng commented on src/main/java/Duke/Parser.java:
```diff
@@ -0,0 +1,99 @@
+    }
+
+    private static boolean isNumber(String str) {
+        if (str == null) return false;
```
Perhaps nest return statement in braces?
```java
if (str == null) {
    return false;
}
```

shaniceng replied:
yes, thanks for highlighting :)

chngchngchng commented on src/main/java/Duke/TaskList.java:
```diff
@@ -0,0 +1,166 @@
+        ArrayList<String> tasksStrings = new ArrayList<String>();
+        for (int i = 0; i < tasks.size(); i++) {
+            Task t = tasks.get(i);
+            String type = t.toString().substring(1,2);
```
Maybe add a small space between numbers and commas?
```suggestion
            String type = t.toString().substring(1, 2);
```

shaniceng replied:
oops, thanks for highlighting 👍

chngchngchng commented on src/main/java/Duke/Ui.java:
```diff
@@ -0,0 +1,119 @@
+     * @return string that returns the final output to be printed for a deleted task
+     */
+    public String printDeleteTask(Task t, int size) {
+        return  String.format("Noted. I've removed this task: \n%s " +
+                "\nNow you have %d tasks in the list.\n", t.toString(), size);
```
Maybe put "+" on a new line?
```suggestion
        return  String.format("Noted. I've removed this task: \n%s " 
        + "\nNow you have %d tasks in the list.\n", t.toString(), size);
```

shaniceng replied:
Thank you for the suggestion :)

chngchngchng commented on src/main/java/Tasks/Task.java:
```diff
@@ -0,0 +1,70 @@
+     * @return String that contains the task name and whether the task is marked or not
+     */
+    public String toString() {
+        return  String.format("%s %s", getStatusIcon(), this.taskName);
```
Extra space?
```suggestion
        return String.format("%s %s", getStatusIcon(), this.taskName);
```

shaniceng replied:
oops, thanks for highlighting!

chngchngchng reviewed (commented):
In general, LGTM! Just some minor issues here and there.

shaniceng committed:
Add assertions to Parser and Storage in Duke

Assert statements used in this program is only to check the length of input by the user.
This helps to test the correctness of any assumptions that have been made in the program.

Let's,
* update the Parser and Storage class to include assert statements so as to ensure our assumptions are correct

shaniceng committed:
Improve Code Quality

Egyptian style braces helps program to have better readability and
placing line breaks at correct position also helps to have better readability
Updated code structures to include level of abstractions.

Let's,
* update the code to use Egyptian style braces
* place line breaks to improve readability

shaniceng committed:
Merge pull request #1 from shaniceng/branch-A-Assertions

Add assertions to Parser and Storage in Duke

shaniceng committed:
Merge branch 'master' of https://github.com/shaniceng/ip

* 'master' of https://github.com/shaniceng/ip:
  Add assertions to Parser and Storage in Duke

shaniceng committed:
Merge pull request #2 from shaniceng/branch-A-CodeQuality

Improve Code Quality

shaniceng committed:
Merge branch 'master' of https://github.com/shaniceng/ip

* 'master' of https://github.com/shaniceng/ip:
  Improve Code Quality

shaniceng committed:
Add C-DetectDuplicates

Detecting duplicates helps to ensure users do not double type the same tasks if they have too many tasks to do

Let's
* update the adding tasks function to detect duplicates by going through the task list
* prints an error message if the exact same tasks has been added in the task list before

shaniceng committed:
Update Javadoc for TaskList

shaniceng committed:
Update Build.gradle

shaniceng committed:
Improve GUI

shaniceng committed:
Add a user guide
Update Find method
Add Help command

shaniceng committed:
Update README.md

shaniceng committed:
Update README.md

shaniceng committed:
Update README.md

shaniceng committed:
Update README.md

shaniceng committed:
Update README.md

shaniceng committed:
Update README.md

shaniceng committed:
Update jar file

shaniceng committed:
Merge branch 'master' of https://github.com/shaniceng/ip

shaniceng committed:
Update README.md

shaniceng committed:
Merge branch 'master' of https://github.com/shaniceng/ip

shaniceng committed:
Update exit platform

shaniceng committed:
Update Help command

shaniceng committed:
Update README.md

shaniceng committed:
Update README.md

shaniceng committed:
Update README.md

shaniceng committed:
Update README.md

Edfernape commented on src/main/java/command/Command.java:
```diff
@@ -0,0 +1,49 @@
+ * This class gets the specific command from the Parser class and
+ * executes the specific command to do the tasks required
+ */
+public abstract class Command {
```
your class hierarchy is amazing

Edfernape reviewed (commented):
Everything looks so organized!
